chameleon-core/src/main/java/org/hibernate/omm/ast/mql/AbstractMQLTranslator.java

```diff
@@ -102,6 +113,21 @@ protected void visitWhereClause(final Predicate whereClauseRestrictions) {
         }
     }
 
+    @Override
+    public void visitNullnessPredicate(NullnessPredicate nullnessPredicate) {
+        Expression expression = nullnessPredicate.getExpression();
+        if (SqlTupleContainer.getSqlTuple(expression) != null) {
+            throw new NotSupportedRuntimeException();
+        }
+
+        String fieldName = mqlAstState.expect(fieldName(), () -> expression.accept(this));
```

Comment:
I think the above usage pattern is not bad and people will get used to it quickly. That being said, I might well explore whether we could solve the issue (basically I think we are taking compiler's responsibility programmatically) during Java compiling phase. For instance, we could insert some well-designed annotation to both expecting statement and returning statement and pre-compiler phase could do the checking automatically for us. Just an idea though.